vm/src/stdlib/time.rs

```diff
@@ -173,6 +197,169 @@ fn time_strptime(
+            }
+            Ok(Duration::new(time.tv_sec as u64, time.tv_nsec as u32))
+        } else if #[cfg(solaris)] {
+            Ok(Duration::from_nanos(unsafe { libc::gethrvtime() }))
+        } else {
+            Err(vm.new_not_implemented_error("thread time unsupported in this system".to_owned()))
+        }
+    }
+}
+
+fn time_thread_time(vm: &VirtualMachine) -> f64 {
+    match get_thread_time(vm) {
+        Ok(v) => v.as_secs_f64(),
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
+
+fn time_thread_time_ns(vm: &VirtualMachine) -> u64 {
+    match get_thread_time(vm) {
+        Ok(v) => v.as_nanos() as u64,
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
+
+#[cfg(windows)]
+fn get_process_time(vm: &VirtualMachine) -> PyResult<Duration> {
+    let mut creation_time = FILETIME::default();
+    let mut exit_time = FILETIME::default();
+    let mut kernel_time = FILETIME::default();
+    let mut user_time = FILETIME::default();
+    unsafe {
+        let process = GetCurrentProcess();
+        if GetProcessTimes(
+            process,
+            &mut creation_time,
+            &mut exit_time,
+            &mut kernel_time,
+            &mut user_time,
+        ) == 0
+        {
+            return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+        }
+
+        let mut large: ULARGE_INTEGER = std::mem::zeroed();
+        large.u_mut().LowPart = kernel_time.dwLowDateTime;
+        large.u_mut().HighPart = kernel_time.dwHighDateTime;
+        let ktime = *large.QuadPart();
+
+        large.u_mut().LowPart = user_time.dwLowDateTime;
+        large.u_mut().HighPart = kernel_time.dwHighDateTime;
+        let utime = *large.QuadPart();
+
+        Ok(Duration::from_nanos((ktime + utime) * 100))
+    }
+}
+
+#[cfg(any(
+    target_os = "wasi",
+    target_os = "redox",
+    target_os = "illumos",
+    target_os = "netbsd",
+    target_os = "solaris"
+))]
+fn time_fromtimeval(tv: libc::timeval, vm: &VirtualMachine) -> PyResult<i64> {
+    let t = match tv.tv_sec.checked_mul(SEC_TO_NS) {
+        Some(value) => value,
+        None => {
+            return Err(vm.new_overflow_error("timestamp too large to convert to i64".to_owned()));
+        }
+    };
+    let u = match tv.tv_usec.checked_mul(US_TO_NS) {
+        Some(value) => value,
+        None => {
+            return Err(vm.new_overflow_error("timestamp too large to convert to i64".to_owned()));
+        }
+    };
+
+    t.checked_add(u)
+        .ok_or_else(|| vm.new_overflow_error("timestamp too large to convert to i64".to_owned()))
+}
+
+#[cfg(not(windows))]
+fn get_process_time(vm: &VirtualMachine) -> PyResult<Duration> {
+    cfg_if::cfg_if! {
+        if #[cfg(any(
+            target_os = "wasi",
+            target_os = "redox",
+            target_os = "illumos",
+            target_os = "netbsd",
+            target_os = "solaris"
+        ))] {
+            let mut ru: libc::rusage = std::mem::zeroed();
```

Comment:
```suggestion
            let mut ru: libc::rusage = unsafe { std::mem::zeroed() };
```
zeroed is unsafe

Reply:
I missed that. Thanks. Is there any way to check grammar works on other platform?